opentracing/src/main/java/io/opentracing/Span.java

```diff
@@ -52,6 +57,14 @@
    * Returns null if no entry found, or baggage is not supported in the current implementation.
    */
   String getBaggageItem(String key);
+  
+  /**
+   * Useful for acquiring the set of baggage keys so that clients of the Span have the ability
+   * to access the corresponding baggage value.
+   * 
+   * @return {@link Set} of baggage keys
+   */
+  Set<String> getBaggageKeys();
```

Comment:
In `Tracer.inject(..)` you're then working with the API at a different level.

It is expected to work more with the internal state.

This internal state is still new, available in https://github.com/opentracing/opentracing-java/pull/20, and it would be appreciated if your efforts could be used in testing and evolving it.